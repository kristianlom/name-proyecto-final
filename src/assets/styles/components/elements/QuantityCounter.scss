@import  '../../lib/palette';

.QuantityCounter {
  width: 86px;
  height: 24px;
  background: map-get(map-get($palette,lightTheme),white);
  border-radius: 12px;
  box-shadow: 0px 0px 4px #00000029;
  opacity: 1;
  display: flex;
  justify-content: space-between;
  button {
    background: none;
  }
  &__icons {
    width: 24px;
    height: 24px;
  }
  &__number {
    font-size: 18px;
    font-family: 'Quicksand Regular';
  }
}