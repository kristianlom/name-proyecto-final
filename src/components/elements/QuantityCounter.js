import React from 'react';
import swal from 'sweetalert';
import { RemoveCircleOutline, AddCircle } from '@material-ui/icons';

import '../../assets/styles/components/elements/QuantityCounter.scss';

const QuantityCounter = (props) => {
  const { action, count, alertCount } = props;

  return (
    <div className="QuantityCounter">
      <button type="button" aria-label="Remove one" onClick={() => { count === 1 ? action(count) : action(count - 1); (count === 1 && alertCount) && swal('You can\'t buy 0 items', 'You can\'t select less than one item', 'error'); }}>
        <RemoveCircleOutline className="QuantityCounter__icons" />
      </button>
      <p className="QuantityCounter__number">{count}</p>
      <button type="button" aria-label="Add one" onClick={() => action(count + 1)} className="QuantityCounter__icons">
        <AddCircle />
      </button>
    </div>
  );
};

export default QuantityCounter;
