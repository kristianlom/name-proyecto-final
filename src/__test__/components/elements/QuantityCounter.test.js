import React from 'react';
import { mount } from 'enzyme';
// import { Box, Icon, Typography } from '@material-ui/core';
import QuantityCounter from '../../../components/elements/QuantityCounter';

describe('Testing render <QuantityCounter /> Component', () => {
  let wrapperComponent;
  beforeEach(() => {
    wrapperComponent = mount(<QuantityCounter />);
  });
  // ---Snapshot:
  it('create snapshot', () => {
    expect(wrapperComponent.find(QuantityCounter)).toMatchSnapshot();
  });
  //  ---Rendering:
  it('component renders correctly', () => {
    expect(wrapperComponent.find(QuantityCounter).length).toEqual(1);
  });
});
