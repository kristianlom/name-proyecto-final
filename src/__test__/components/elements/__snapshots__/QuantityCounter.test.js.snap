// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render <QuantityCounter /> Component create snapshot 1`] = `ReactWrapper {}`;
